modules/lib/shared/content.d.ts

```diff
@@ -0,0 +1,63 @@
+declare global {
+    // eslint-disable-next-line @typescript-eslint/no-empty-interface
+    interface XpXData {
+    }
+}
+
+import {Component, Region} from './portal';
+
+export type WorkflowState = 'IN_PROGRESS' | 'PENDING_APPROVAL' | 'REJECTED' | 'READY';
+
+export type WorkflowCheckState = 'PENDING' | 'REJECTED' | 'APPROVED';
+
+export type ContentInheritType = 'CONTENT' | 'PARENT' | 'NAME' | 'SORT';
+
+export interface Workflow {
+    state: WorkflowState;
+    checks?: Record<string, WorkflowCheckState>;
+}
+
+export interface Attachment {
+    name: string;
+    label?: string;
+    size: number;
+    mimeType: string;
+}
+
+export interface PublishInfo {
+    from?: string;
+    to?: string;
+    first?: string;
+}
+
+export interface Content<
+    Data = Record<string, unknown>,
+    Type extends string = string,
+    Config extends object = object,
+    Regions extends Record<string, Region> = Record<string, Region>,
```

Comment:
I am a little bit unsure if `Config` and `Regions` should always be type parameters but lets talk about that.